src/SFML/Window/Android/JniHelper.cpp

```diff
@@ -242,39 +242,49 @@ std::optional<JniList<JniMotionRange, JniMotionRangeClass>> JniInputDevice::getM
 
 
 ////////////////////////////////////////////////////////////
-Jni::Jni(Jni&& other) noexcept
+Jni::Jni(JavaVM& vm, JNIEnv& env) : m_vm(&vm), m_env(&env)
 {
-    std::swap(m_vm, other.m_vm);
 }
 
 
 ////////////////////////////////////////////////////////////
-Jni::Jni(JavaVM* vm) : m_vm(vm)
+Jni::Jni(Jni&& other) noexcept
 {
+    std::swap(m_vm, other.m_vm);
+    std::swap(m_env, other.m_env);
 }
 
 
 ////////////////////////////////////////////////////////////
 Jni::~Jni()
 {
+    m_env = nullptr;
     if (m_vm)
```

Comment:
Is there a situation where `m_vm` would be null upon destruction?

Reply:
> the moved-from object will contain nulls and trigger the assert.

This is not necessarily true, but it's certainly possible. We can't know the value of `m_vm` when it's in a moved-from state. The broader question is, does this class really need to be moveable? I noticed you added some special members while deleting others. Can you talk me through your rationale there? In general I'd prefer to disable as many potentially problematic special members functions as possible if we don't need them.